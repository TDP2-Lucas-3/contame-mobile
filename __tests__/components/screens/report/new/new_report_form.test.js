import React from 'react';
import {shallow} from 'enzyme';
import NewReportForm from '../../../../../src/components/screens/reports/new/new_report_form';

describe('NewReportForm', () => {
  let render;
  let props;

  beforeEach(() => {
    props = {
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      loading: false,
      data: {},
    };
    render = (appProps) => shallow(<NewReportForm {...appProps} />);
  });

  it('renders without crashing', () => {
    expect(() => render(props)).not.toThrowError();
  });

  it('renders all values correctly', () => {
    expect(render(props)).toMatchSnapshot();
  });
});
