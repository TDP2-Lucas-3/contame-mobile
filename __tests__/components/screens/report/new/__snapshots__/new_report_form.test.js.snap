// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewReportForm renders all values correctly 1`] = `
<View>
  <ForwardRef
    nextStepAnimation="slideRight"
    prevStepAnimation="slideLeft"
    steps={
      Array [
        Object {
          "content": <CategoryStep
            onChange={[Function]}
          />,
        },
        Object {
          "content": <TitleStep
            onBack={[Function]}
            onChange={[MockFunction]}
            onSelect={[Function]}
          />,
        },
        Object {
          "content": <DescriptionStep
            onBack={[Function]}
            onChange={[MockFunction]}
            onSelect={[Function]}
          />,
        },
        Object {
          "content": <IncidentLocationMapViewStep
            onSelect={[Function]}
            skip={[Function]}
          />,
        },
        Object {
          "content": <ImagesStep
            onChange={[MockFunction]}
            onNext={[Function]}
          />,
        },
        Object {
          "content": <ConfirmStep
            data={Object {}}
            first={[Function]}
            onSubmit={[MockFunction]}
          />,
        },
      ]
    }
  />
</View>
`;
